### PR TITLE
Federated Clients: register federated client via MarioClient and clean up interface

### PR DESCRIPTION
**Pull request by jonlee2**

### CODE REVIEW

File: li-apache-kafka-clients/src/test/java/com/linkedin/kafka/clients/metadataservice/MarioMetadataServiceClientTest.java [hunk 1]
```diff
@@ -70,7 +73,23 @@ public boolean matches(TopicQuery other) {
   public void setup() {
     _marioClient = Mockito.mock(MarioClient.class);
     _marioMetadataServiceClient = new MarioMetadataServiceClient(_marioClient);
-    _clientId = _marioMetadataServiceClient.registerFederatedClient(CLUSTER_GROUP, null, 1000);
+  }
+
+  @Test
+  public void testRegusterFederatedClient() throws MetadataServiceClientException {
```
Comment (kehuum) [inline li-apache-kafka-clients/src/test/java/com/linkedin/kafka/clients/metadataservice/MarioMetadataServiceClientTest.java]: typo, testRegisterFederatedClient

Comment (jonlee2) [reply li-apache-kafka-clients/src/test/java/com/linkedin/kafka/clients/metadataservice/MarioMetadataServiceClientTest.java]: Fixed

File: li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/common/FederatedClientCommandCallback.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.UUID;
+
+
+// A callback interface for executing commands requested by the metadata service.
+//
+// A command-specific callback will implement this interface.
+public interface FederatedClientCommandCallback {
+  public FederatedClientCommandType getCommandType();
+
+  public void onReceivingCommand(UUID commandRequestId, Map<String, String> args);
```
Comment (radai-rosenblatt) [inline li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/common/FederatedClientCommandCallback.java]: nit - methods in interfaces are public anyway

File: li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/metadataservice/MarioCommandCallbackImpl.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      _federatedClientCommandCallbacks.put(callback.getCommandType(), callback);
+    }
+  }
+
+  public void onReceivingCommand(Messages marioCommandMessage) {
+    // Find a federate client callback that matches the given Mario command message type and execute it with arguments
+    // included in the message.
+    switch (marioCommandMessage.getMsgType()) {
+      default:
+        // No current support at the moment
```
Comment (radai-rosenblatt) [inline li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/metadataservice/MarioCommandCallbackImpl.java]: default should at least print a warn

File: li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/common/FederatedClientCommandCallback.java (hunk 2 above)
Comment (jonlee2) [reply li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/common/FederatedClientCommandCallback.java]: removed public

File: li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/metadataservice/MarioCommandCallbackImpl.java (hunk 3 above)
Comment (jonlee2) [reply li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/metadataservice/MarioCommandCallbackImpl.java]: added a warning

**Pull request merged**